test/buglistTests.js

```diff
@@ -27,19 +31,108 @@ while ((result = testVectorParser.exec(tests)) !== null)
     var fine = result[3].split('\n--\n')
     console.log("Testing " + name + " with " + buggy.length + " buggy and " + fine.length + " fine instances")
 
-    var regex = RegExp(bugsByName[name].check['regex-source'])
-    for (var i in buggy)
+	checkRegex(name, buggy, fine)
+	checkJSONPath(name, buggy, fine)
+}
+
+function checkRegex(name, buggy, fine)
+{
+    var regexStr = bugsByName[name].check['regex-source']
+    if (regexStr !== undefined)
     {
-        if (!regex.exec(buggy[i]))
+        var regex = RegExp(regexStr)
+        for (var i in buggy)
         {
-            throw "Bug " + name + ": Buggy source does not match: " + buggy[i]
+            if (!regex.exec(buggy[i]))
+            {
+                throw "Bug " + name + ": Buggy source does not match: " + buggy[i]
+            }
         }
-    }
-    for (var i in fine)
-    {
-        if (regex.exec(fine[i]))
+        for (var i in fine)
         {
-            throw "Bug " + name + ": Non-buggy source matches: " + fine[i]
+            if (regex.exec(fine[i]))
+            {
+                throw "Bug " + name + ": Non-buggy source matches: " + fine[i]
+            }
         }
     }
 }
+
+function checkJSONPath(name, buggy, fine)
+{
+    var jsonPath = bugsByName[name].check['json-path']
+    if (jsonPath !== undefined)
+    {
+        var url = "http://github.com/ethereum/solidity/releases/download/v" + bugsByName[name].introduced + "/solc-static-linux"
+        var binary = __dirname + "/solc-static-linux"
+        download(url, __dirname)
+        .then(() => {
```

Comment:
Doing this async will interfere with the logging in the main context. Can you add some "join" call at the end or whatever the equivalent is?